### PR TITLE
Improve documentation.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: doc/TROUBLESHOOT.md [hunk 1]
```diff
@@ -4,24 +4,51 @@ The text below is under construction.
...
 TODO: separate information for tests and applications.
 
-# Memory Leak Troubleshooting
+# Troubleshoot Memory Leaks
 
-This page describes how to troubleshoot memory leaks. See leak detection guidance [here](DETECT.md).
+This page describes how to troubleshoot memory leaks. See other information on memory leaks [here](../README.md).
 
-## Collect callstack
+To understand root cause of a memroy leak, gather additional information.
```
Comment (kenzieschmoll) [inline doc/TROUBLESHOOT.md:11]: sp: memory

To understand the root cause of a memory leak, you may need to gather additional information.

File: doc/TROUBLESHOOT.md [hunk 2]
```diff
@@ -4,24 +4,51 @@ The text below is under construction.
...
-the leak's root cause. The lifecycle event will be creation
-for not-disposed leaks, and disposal for non-GCed leaks.
+- **not-disposed**:
+
+    - **Allocation call-stack** helps to detect
+        owner of the object, that is responsible for the object disposal.
+
+- **not-GCed or GCed-late**:
+
+    - **Allocation and disposal call-stacks**: help to understand lifecicle of the
```
Comment (kenzieschmoll) [inline doc/TROUBLESHOOT.md:20]: r/help/helps
sp: lifecycle
'object, which may reveal where the object is being held from garbage collection'

**Review by kenzieschmoll — COMMENTED** [review]
some wording suggestions

File: doc/CONCEPTS.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    - **Definition**: an object was disposed and then GCed,
+       but GC happened later than expected. This means the retaining path was
+       holding the object in memory for some period, but then disappeared.
+
+    - **Fix**: the same as for **not-GCed**
+
+- **Disposed, but not GCed, without path (not-GCed-without-path)**:
+    - **Definition**: an object
+       was disposed and not GCed when expected, but retaining path
+       is not detected,
```
[the remark this answers is not in the document]
Comment (polina-c) [reply doc/CONCEPTS.md]: It is supposed to be one sentence. Replaced ',' with ';'

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**